### PR TITLE
Media keys fixes and workarounds

### PR DESCRIPTION
**Pull request by JasonLG1979**

### CODE REVIEW

File: pithos/plugins/mediakeys.py [hunk 1]
```diff
@@ -92,13 +97,15 @@ def prepare_keybinder():
                     self.prepare_complete()
 
         def on_new_finish(source, result, data):
+            nonlocal de
```
Comment (TingPing) [inline pithos/plugins/mediakeys.py]: I'm lost, where is `de` actually set?

Comment (JasonLG1979) [reply pithos/plugins/mediakeys.py]: https://github.com/JasonLG1979/pithos/blob/876e08a60d8541b96318fa4d6e33f966e09a8db5/pithos/plugins/mediakeys.py#L134 or https://github.com/JasonLG1979/pithos/blob/876e08a60d8541b96318fa4d6e33f966e09a8db5/pithos/plugins/mediakeys.py#L112

**Comment by JasonLG1979** [comment]
I would like this included in 1.4.1. It's about as good as it's going to get as far as making DBus mediakeys function as they should, at least we do our part. As I've said It's up to the DE as far as what they do with the mediakeys. But we grab them when we should...

File: pithos/plugins/mediakeys.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def mediakey_signal(self, proxy, sender, signal, param, userdata=None):
                 self.window.playpause_notify()
             elif action == 'Next':
                 self.window.next_song()
-            elif action == 'Stop':
+            elif action == 'Stop' or 'Pause':
```
Comment (TingPing) [inline pithos/plugins/mediakeys.py]: This isn't valid syntax?

Comment (ForLoveOfCats) [reply pithos/plugins/mediakeys.py]: It should be:
```elif action == 'Stop' or action == 'Pause':```

Comment (JasonLG1979) [reply pithos/plugins/mediakeys.py]: Yes it is.

Comment (JasonLG1979) [reply pithos/plugins/mediakeys.py]: It works as expected.

Comment (JasonLG1979) [reply pithos/plugins/mediakeys.py]: if the action equals/is 'Stop' or 'Pause' user_pause()

Comment (JasonLG1979) [reply pithos/plugins/mediakeys.py]: Well my bad it's not technically invalid syntax but it does not work as I expected. I'll fix it.

**Comment by JasonLG1979** [comment]
@TingPing There fixed ;)

File: pithos/plugins/mediakeys.py [hunk 3]
```diff
@@ -68,10 +73,10 @@ def mediakey_signal(self, proxy, sender, signal, param, userdata=None):
                 self.window.playpause_notify()
             elif action == 'Next':
                 self.window.next_song()
-            elif action == 'Stop' or 'Pause':
-                self.window.user_pause()
             elif action == 'Previous':
                 self.window.bring_to_top()
+            elif action in ('Stop', 'Pause'):
```
Comment (TingPing) [inline pithos/plugins/mediakeys.py]: Well now you got your commits messed up, this needs to be rebased into the first one.

Comment (JasonLG1979) [reply pithos/plugins/mediakeys.py]: So just squash it before you push it. Both commit titles will be in the squashed commit message.

Comment (TingPing) [reply pithos/plugins/mediakeys.py]: There is a purpose to preserving history of unrelated bug-fixes.

Comment (JasonLG1979) [reply pithos/plugins/mediakeys.py]: OK I'll work my git magik,lol!!!

Comment (TingPing) [reply pithos/plugins/mediakeys.py]:
```sh
git reset HEAD~2
git add -p
# select just the one hunk
git commit -m 'mediakeys: Listen for 'Pause' mediakeys action'
git commit -am 'mediakeys: Workaround MATE not updating it's window state properly'
```

**Comment by JasonLG1979** [comment]
Done.

**Comment by JasonLG1979** [comment]
That last commit simplifies the logic a bit. IMHO it makes it a little easier to follow.

**Comment by TingPing** [comment]
@JasonLG1979 Look good?:

```xml
    <release date="2017-11-26" version="1.4.1">
      <description>
        <p>This is a minor release fixing up some bugs:</p>
        <ul>
          <li>Change default quality to high (mp3)</li>
          <li>Mark application as DBusActivatable</li>
          <li>MPRIS: Fix potential unhandled exception</li>
          <li>Mediakeys: Handle keyboards with a dedicated pause key</li>
          <li>Mediakeys: Fix support on GNOME-Shell 3.26+ and MATE</li>
          <li>Notification Icon: Improve behavior on various notification servers</li>
        </ul>
      </description>
    </release>
```

**Comment by JasonLG1979** [comment]
Looks good to me ;)

**Pull request merged**